src/utils/useStateReadState/index.ts

```diff
@@ -1,123 +1,45 @@
+  const [data, setData] = useState<LotusActorState<T> | null>(null)
+  const [loading, setLoading] = useState<boolean>(false)
+  const [notFound, setNotFound] = useState<boolean>(false)
+  const [error, setError] = useState<Error | null>(null)
+  const { lotusApi } = useEnvironment()
 
   useEffect(() => {
-    const fetchState = async () => {
-      try {
-        const res = await lotusApi.request<LotusRPCActorState<any>>(
-          'StateReadState',
-          baseOptions.variables.address,
-          null
-        )
-
-        if (res && decodeActorCID(res.Code, networkName).includes('multisig')) {
-          res.State = res.State as MsigState<string>
-          const signers = await Promise.all(
-            res.State.Signers.map(async (id: string) => {
-              const { data } = await apolloClient.query<AddressQuery>({
-                query: AddressDocument,
-                variables: {
-                  address: id
-                }
-              })
-
-              return { id, robust: data?.address?.robust || '' } as Address
-            })
-          )
-
-          const availableBalance = await lotusApi.request<string>(
-            'MsigGetAvailableBalance',
-            baseOptions.variables.address,
-            null
-          )
-
-          setActorState({
-            ...res,
-            State: {
-              ...res.State,
-              Signers: signers,
-              AvailableBalance: availableBalance
-            }
-          })
-        } else if (res) {
-          setActorState({ ...res })
-        }
-      } catch (err) {
-        if (err instanceof Error) {
-          if (err.message.includes('actor not found')) {
-            setActorState({ ...emptyActorState })
-          } else setError(err)
-        } else setError(new Error('Failed to fetch'))
-      } finally {
-        setLoading(false)
-      }
-    }
-
-    const firstLoad = !actorState && loading && !error
-    const newLoad = fetchedFor !== baseOptions?.variables?.address && !error
-
-    if (firstLoad || newLoad) {
+    setData(null)
+    setNotFound(false)
+    setError(null)
+    if (!address) return
+    if (validateAddressString(address)) {
       setLoading(true)
-      setFetchedFor(baseOptions?.variables?.address)
-      fetchState()
+      lotusApi
+        .request<LotusActorState<T>>('StateReadState', address, null)
+        .then((a: LotusActorState<T>) => setData(a))
+        .catch((e: Error) => {
+          e.message.includes('actor not found')
+            ? setNotFound(true)
+            : setError(e)
+        })
+        .finally(() => setLoading(false))
+    } else {
+      setError(new Error('Invalid address'))
     }
-  }, [
-    baseOptions?.variables?.address,
-    loading,
-    error,
-    actorState,
-    fetchedFor,
-    setFetchedFor,
-    apolloClient,
-    lotusApi,
-    networkName
-  ])
+  }, [address, lotusApi])
 
-  return { data: actorState, error, loading }
+  return { data, loading, notFound, error }
```

Comment:
Return a `notFound` property instead of an empty object. Then we can show in the frontend what is going on instead of displaying an actor page with empty values. I was confused about this before.